internal/controllers/machineset/machineset_controller.go

```diff
@@ -969,12 +994,47 @@ func (r *Reconciler) getMachineNode(ctx context.Context, cluster *clusterv1.Clus
 
 func (r *Reconciler) reconcileUnhealthyMachines(ctx context.Context, cluster *clusterv1.Cluster, ms *clusterv1.MachineSet, filteredMachines []*clusterv1.Machine) (ctrl.Result, error) {
 	log := ctrl.LoggerFrom(ctx)
+
```

Comment:
should we return quickly if there are no machines to be remediated?

Reply:
We're returning as soon as we get the max in flight data from the MachineDeployment a few lines down